scifin/timeseries/timeseries.py

```diff
@@ -79,55 +79,49 @@ class Series:
       Timezone associated with dates.
     """
     
-    def __init__(self, df=None, tz=None, unit=None, name=""):
+    def __init__(self, data=None, tz=None, unit=None, name=""):
         """
-        Receives a data frame as an argument and initializes the time series.
+        Receives a panda.Series or pandas.DataFrame as an argument and initializes the time series.
         """
         
-        # Deal with DataFrame
-        if (df is None) or (df.empty == True):
-            self.data = pd.DataFrame(index=None, data=None)
+        # Deal with DataFrame / Series
+        if (data is None) or (data.empty == True):
+            self.data = pd.Series(index=None, data=None)
             self.start_utc = None
             self.end_utc = None
             self.nvalues = 0
             self.freq = None
             self.name = 'Empty TimeSeries'
         else:
-            # Making sure the dataframe is just
-            # an index + 1 value column
-            try:
-                assert(df.shape[1]==1)
-            except AssertionError:
-                raise AssertionError("Time series must be built from a data frame with only one value column.")
-                         
-            # Extract values
-            if type(df.index[0]) == 'str':
-                new_index = pd.to_datetime(df.index, format=fmt)
-                self.data = pd.DataFrame(index=new_index, data=df.values)
+            # Making sure the user entered a pandas.Series or pandas.DataFrame
+            # with just an index and one column for values
+            if isinstance(data, pd.DataFrame):
+                if data.shape[1]!=1:
+                    raise AssertionError("Time series must be built from a pandas.Series or a pandas.DataFrame with only one value column.")
+                else:
+                    self.data = pd.Series(data.iloc[:,0])
+            elif isinstance(data, pd.Series) == False:
```

Comment:
I would prefer `` not isinstance(data, pd.Series)`` instead of checking against false

Reply:
Corrected.